REQUIRE

```diff
@@ -4,7 +4,7 @@ Distributions
 Docile
 DualNumbers
 FITSIO
-ForwardDiff 0.0.3 0.1
+ForwardDiff # Currently only required for testing.
```

Comment:
That doesn't look so bad, if we're just using ForwardDiff.jl for testing
rather than GSL.jl. Or did you have in mind using ForwardDiff.jl outside of
the test directory?

On Thu, Nov 19, 2015 at 9:56 PM Ryan <notifications@github.com> wrote:

> In REQUIRE
> <https://github.com/jeff-regier/Celeste.jl/pull/89#discussion_r45438141>:
>
> > @@ -4,7 +4,7 @@ Distributions
> >  Docile
> >  DualNumbers
> >  FITSIO
> > -ForwardDiff 0.0.3 0.1
> > +ForwardDiff # Currently only required for testing.
>
> Unfortunately, ForwardDiff types allocate a ton of memory. We can't use
> them as a drop-in replacement for our own derivatives or Hessians. It might
> be good to point this out to Miles, though. He might have some ideas.
>
> # ForwardDiff types:
> julia> @time for n=1:10000
>          foo = sum([ mp_fd.vp[1][i] * mp_fd.vp[1][i] for i=1:length(ids)] )
>        end
>   0.306251 seconds (2.26 M allocations: 592.808 MB, 23.21% gc time)
>
> # Ordinary types:
> julia> @time for n=1:10000
>          bar = sum([ mp.vp[1][i] * mp.vp[1][i] for i=1:length(ids)] )
>        end
>   0.093861 seconds (1.29 M allocations: 22.739 MB, 4.02% gc time)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jeff-regier/Celeste.jl/pull/89/files#r45438141>.
>